src/js/App/Sidenav/Navigation/DynamicNav.js

```diff
@@ -1,57 +1,58 @@
+import isEqual from 'lodash/isEqual';
+import ChromeNavItem from './ChromeNavItem';
 import { loadLeftNavSegment } from '../../../redux/actions';
 
-const DynamicNav = ({ dynamicNav }) => {
-  const {
-    location: { pathname },
-  } = useHistory();
-  const dispatch = useDispatch();
-  const [appName] = dynamicNav.split('/');
+const toArray = (value) => (Array.isArray(value) ? value : [value]);
+const mergeArrays = (orig, index, value) => [...orig.slice(0, index), ...toArray(value), ...orig.slice(index)];
+
+const HookedNavigation = ({ useNavigation, dynamicNav, pathname, ...props }) => {
   const currentNamespace = pathname.split('/')[1];
+  const [isLoaded, setIsLoaded] = useState(false);
+  const dispatch = useDispatch();
   const schema = useSelector(({ chrome: { navigation } }) => navigation[currentNamespace]);
-  const [{ default: navigation }] = useLoadModule({ appName, scope: appName, module: './Navigation' }, {});
+  const currNav = useSelector(({ chrome: { navigation } }) =>
+    navigation[currentNamespace]?.navItems?.filter((item) => item.dynamicNav === dynamicNav)
+  );
+  const newNav = useNavigation({ schema, dynamicNav, currentNamespace, currNav });
   useEffect(() => {
-    if (navigation) {
-      if (typeof navigation === 'function') {
-        const indexOfDynamicNav = schema.navItems.findIndex((nav) => nav.dynamicNav === dynamicNav);
-        if (indexOfDynamicNav !== -1) {
-          const { dynamicNav: _dynamicNav, ...originalNav } = schema.navItems[indexOfDynamicNav];
-          Promise.resolve(navigation({ schema, dynamicNav, currentNamespace })).then((data) => {
-            const newValue = Array.isArray(data)
-              ? data.map((item) => ({
-                  appId: dynamicNav.split('/')[0],
-                  ...originalNav,
-                  ...item,
-                }))
-              : [
-                  {
-                    ...originalNav,
-                    ...data,
-                  },
-                ];
-            dispatch(
-              loadLeftNavSegment(
-                {
-                  ...schema,
-                  navItems: schema.navItems.flatMap((item, key) => (key === indexOfDynamicNav ? newValue : item)),
-                },
-                currentNamespace,
-                pathname,
-                true
-              )
-            );
-          });
-        }
+    if (newNav?.length > 0) {
+      const newValue = toArray(newNav).map((item, key) => ({
+        appId: dynamicNav.split('/')[0],
+        ...(currNav?.[key] || currNav?.[0]),
+        ...item,
+      }));
+      if (!isEqual(newValue, currNav)) {
+        const currNavIndex = schema.navItems.findIndex((item) => item.dynamicNav === dynamicNav);
+        dispatch(
+          loadLeftNavSegment(
+            {
+              ...schema,
+              navItems: mergeArrays(
+                schema.navItems.filter((item) => !(item.dynamicNav && item.dynamicNav === dynamicNav)),
+                currNavIndex,
```

Comment:
If `currNavIndex` is `-1`, your array splicing is going to get weird. Did you want to safeguard against the unable to find `findIndex` call above?

Reply:
Yeah, good point. It shouldn't happen since we took the dynamic nav from the navigation itself. Let's be defensive and safeguard it!